<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ SplitClusterTest</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SplitClusterTest</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/one-cell-type.html">One Cell Type</a></li>
    <li><a class="dropdown-item" href="../articles/two-cell-types.html">Two Cell Types</a></li>
    <li><a class="dropdown-item" href="../articles/two-gaussians.html">Two Gaussians</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/szcf-weiya/SplitClusterTest/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>



      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="calc_acc.html">calc_acc()</a></code>

        </dt>
        <dd>Calculate the accuracy (FDR, Power, F1 score) if truth is available</dd>

        <dt>

          <code><a href="calc_inc_rate.html">calc_inc_rate()</a></code>

        </dt>
        <dd>Calculate the inclusion rate</dd>

        <dt>

          <code><a href="calc_tau.html">calc_tau()</a></code>

        </dt>
        <dd>Calculate the cutoff the mirror statistics</dd>

        <dt>

          <code><a href="cluster_diff.html">cluster_diff()</a></code>

        </dt>
        <dd>Calculate the difference across two clusters for each feature</dd>

        <dt>

          <code><a href="dd.html">dd()</a></code>

        </dt>
        <dd>Wrapper for the naive double-dipping method</dd>

        <dt>

          <code><a href="debias_symmetry.html">debias_symmetry()</a></code>

        </dt>
        <dd>Debias the statistics under the null for symmetry</dd>

        <dt>

          <code><a href="ds.html">ds()</a></code>

        </dt>
        <dd>DS procedure</dd>

        <dt>

          <code><a href="ds.SingleCellExperiment.html">ds(<i>&lt;SingleCellExperiment&gt;</i>)</a></code>

        </dt>
        <dd>DS procedure for Seurat object</dd>

        <dt>

          <code><a href="ds.matrix.html">ds(<i>&lt;matrix&gt;</i>)</a></code>

        </dt>
        <dd>DS procedure for Matrix</dd>

        <dt>

          <code><a href="est.Sigma.html">est.Sigma()</a></code>

        </dt>
        <dd>estimate the covariance matrix (assuming there are two clusters)</dd>

        <dt>

          <code><a href="gen_data_normal.html">gen_data_normal()</a></code>

        </dt>
        <dd>Generate simulation data with two gaussians</dd>

        <dt>

          <code><a href="mds.html">mds()</a></code>

        </dt>
        <dd>MDS procedure</dd>

        <dt>

          <code><a href="mds.SingleCellExperiment.html">mds(<i>&lt;SingleCellExperiment&gt;</i>)</a></code>

        </dt>
        <dd>Multiple data splitting</dd>

        <dt>

          <code><a href="mds.matrix.html">mds(<i>&lt;matrix&gt;</i>)</a></code>

        </dt>
        <dd>MDS procedure for matrix</dd>

        <dt>

          <code><a href="mds1.html">mds1()</a></code>

        </dt>
        <dd>Conduct multiple data splitting</dd>

        <dt>

          <code><a href="mds1_parallel.html">mds1_parallel()</a></code>

        </dt>
        <dd>Conduct multiple data splitting in parallel</dd>

        <dt>

          <code><a href="mds2.html">mds2()</a></code>

        </dt>
        <dd>Aggregate multiple data splitting results, and return a selection set</dd>

        <dt>

          <code><a href="mirror_stat.html">mirror_stat()</a></code>

        </dt>
        <dd>Calculate the mirror statistics</dd>

        <dt>

          <code><a href="myDiffTTest.html">myDiffTTest()</a></code>

        </dt>
        <dd>Modified Seurat::DiffTTest by extending the output with statistics in addition to p-values</dd>

        <dt>

          <code><a href="myFindMarkers.html">myFindMarkers()</a></code>

        </dt>
        <dd>Gene expression markers of identity classes</dd>

        <dt>

          <code><a href="myGLMDETest.html">myGLMDETest()</a></code>

        </dt>
        <dd>Modified Seurat::GLMDETest by extending the output with statistics in addition to p-values</dd>

        <dt>

          <code><a href="myWilcoxDETest.html">myWilcoxDETest()</a></code>

        </dt>
        <dd>Modified Seurat::WilcoxDETest by extending the output with statistics in addition to p-values</dd>

        <dt>

          <code><a href="perform_clustering.html">perform_clustering()</a></code>

        </dt>
        <dd>Perform Clustering on a SingleCellExperiment object</dd>

        <dt>

          <code><a href="rankSumTestWithCorrelation.html">rankSumTestWithCorrelation()</a></code>

        </dt>
        <dd>Wilcoxon rank sum test (adapted from <code>limma::rankSumTestWithCorrelation</code>)</dd>

        <dt>

          <code><a href="sel_inc_rate.html">sel_inc_rate()</a></code>

        </dt>
        <dd>Perform selection based on inclusion rate</dd>

        <dt>

          <code><a href="simdata_1ct.html">simdata_1ct</a></code>

        </dt>
        <dd>Datasets Demo synthetic scRNA-seq data with one cell type based on <code>DuoClustering2018::sce_full_Zhengmix4eq()</code></dd>

        <dt>

          <code><a href="simdata_2ct.html">simdata_2ct</a></code>

        </dt>
        <dd>Demo synthetic scRNA-seq data with two cell types based on <code>DuoClustering2018::sce_full_Zhengmix4eq()</code></dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://hohoweiya.xyz" class="external-link">Lijun Wang</a>, <a href="https://yingxinlin.github.io/" class="external-link">Yingxin Lin</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

