<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>One Cell Type â€¢ SplitClusterTest</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="One Cell Type">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SplitClusterTest</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/one-cell-type.html">One Cell Type</a></li>
    <li><a class="dropdown-item" href="../articles/two-cell-types.html">Two Cell Types</a></li>
    <li><a class="dropdown-item" href="../articles/two-gaussians.html">Two Gaussians</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/szcf-weiya/SplitClusterTest/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>One Cell Type</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/szcf-weiya/SplitClusterTest/blob/v0.1.1/vignettes/one-cell-type.Rmd" class="external-link"><code>vignettes/one-cell-type.Rmd</code></a></small>
      <div class="d-none name"><code>one-cell-type.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/szcf-weiya/SplitClusterTest" class="external-link">SplitClusterTest</a></span><span class="op">)</span></span></code></pre></div>
<p>The simulated data <code>simdata_1ct</code> with one cell type (i.e.,
no cluster structure) is generated based on the real single-cell data
<code>DuoClustering2018::sce_full_Zhengmix4eq()</code> with the help of
<code>scDesign3</code>. For more details about generating synthetic
data, please check our paper.</p>
<p>The structure of <code>simdata_1ct</code> is as follows:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">simdata_1ct</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 2</span></span>
<span><span class="co">#&gt;  $ simu_sce:Formal class 'SingleCellExperiment' [package "SingleCellExperiment"] with 9 slots</span></span>
<span><span class="co">#&gt;   .. ..@ int_elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ nrows          : int 198</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ listData       :List of 1</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ rowPairs:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ nrows          : int 198</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. ..@ int_colData        :Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ nrows          : int 998</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ listData       :List of 3</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ reducedDims:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ nrows          : int 998</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ altExps    :Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ nrows          : int 998</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ colPairs   :Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ nrows          : int 998</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. ..@ int_metadata       :List of 1</span></span>
<span><span class="co">#&gt;   .. .. ..$ version:Classes 'package_version', 'numeric_version'  hidden list of 1</span></span>
<span><span class="co">#&gt;   .. .. .. ..$ : int [1:3] 1 24 0</span></span>
<span><span class="co">#&gt;   .. ..@ rowRanges          :Formal class 'CompressedGRangesList' [package "GenomicRanges"] with 5 slots</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ unlistData     :Formal class 'GRanges' [package "GenomicRanges"] with 7 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ seqnames       :Formal class 'Rle' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ values         : Factor w/ 0 levels: </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ lengths        : int(0) </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ ranges         :Formal class 'IRanges' [package "IRanges"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ start          : int(0) </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ width          : int(0) </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ NAMES          : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ strand         :Formal class 'Rle' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ values         : Factor w/ 3 levels "+","-","*": </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ lengths        : int(0) </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package "GenomeInfoDb"] with 4 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ seqnames   : chr(0) </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ seqlengths : int(0) </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ is_circular: logi(0) </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ genome     : chr(0) </span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ nrows          : int 0</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementMetadata:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ nrows          : int 198</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementType    : chr "GRanges"</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ partitioning   :Formal class 'PartitioningByEnd' [package "IRanges"] with 5 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ end            : int [1:198] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ NAMES          : chr [1:198] "ENSG00000116251" "ENSG00000142676" "ENSG00000142669" "ENSG00000169442" ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. ..@ colData            :Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ rownames       : chr [1:998] "naive.cytotoxic10013" "naive.cytotoxic5827" "naive.cytotoxic1319" "naive.cytotoxic4199" ...</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ nrows          : int 998</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ listData       :List of 1</span></span>
<span><span class="co">#&gt;   .. .. .. .. ..$ cell_type: chr [1:998] "naive.cytotoxic" "naive.cytotoxic" "naive.cytotoxic" "naive.cytotoxic" ...</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. ..@ assays             :Formal class 'SimpleAssays' [package "SummarizedExperiment"] with 1 slot</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ data:Formal class 'SimpleList' [package "S4Vectors"] with 4 slots</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ listData       :List of 2</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..$ counts   : num [1:198, 1:998] 3 19 2 5 1 6 0 12 1 3 ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. .. ..$ : chr [1:198] "ENSG00000116251" "ENSG00000142676" "ENSG00000142669" "ENSG00000169442" ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. .. ..$ : chr [1:998] "naive.cytotoxic10013" "naive.cytotoxic5827" "naive.cytotoxic1319" "naive.cytotoxic4199" ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. ..$ logcounts: num [1:198, 1:998] 1.386 2.996 1.099 1.792 0.693 ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. .. ..$ : chr [1:198] "ENSG00000116251" "ENSG00000142676" "ENSG00000142669" "ENSG00000169442" ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. .. .. .. ..$ : chr [1:998] "naive.cytotoxic10013" "naive.cytotoxic5827" "naive.cytotoxic1319" "naive.cytotoxic4199" ...</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. ..@ NAMES              : NULL</span></span>
<span><span class="co">#&gt;   .. ..@ elementMetadata    :Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ rownames       : NULL</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ nrows          : int 198</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ listData       : Named list()</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementType    : chr "ANY"</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ elementMetadata: NULL</span></span>
<span><span class="co">#&gt;   .. .. .. ..@ metadata       : list()</span></span>
<span><span class="co">#&gt;   .. ..@ metadata           : list()</span></span>
<span><span class="co">#&gt;  $ de_idx  : NULL</span></span></code></pre></div>
<p>Since there is no cluster structure, so the index of DE genes is
empty:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">simdata_1ct</span><span class="op">$</span><span class="va">de_idx</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
<p>Our proposed multiple data splitting (MDS) does not return
significant DE genes:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mss</span> <span class="op">=</span> <span class="fu"><a href="../reference/mds1.html">mds1</a></span><span class="op">(</span><span class="va">simdata_1ct</span><span class="op">$</span><span class="va">simu_sce</span>, M <span class="op">=</span> <span class="fl">1</span>, </span>
<span>           params1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>normalized_method <span class="op">=</span> <span class="st">"sct"</span>, pca.whiten <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>           params2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>normalized_method <span class="op">=</span> <span class="st">"sct"</span>, pca.whiten <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ===== Multiple Data Splitting:  1 / 1  =====</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ----- data splitting (1st half) -----</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ----- data splitting (2nd half) ----</span></span>
<span><span class="va">sel</span> <span class="op">=</span> <span class="fu"><a href="../reference/mds2.html">mds2</a></span><span class="op">(</span><span class="va">mss</span><span class="op">)</span></span></code></pre></div>
<p>The mirror statistics are distributed as follows:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">mss</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, breaks <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span></code></pre></div>
<p><img src="one-cell-type_files/figure-html/hist-mirror-one-1.png" class="r-plt" alt="Histogram of Mirror Statistics (One Cell Type)" width="700"></p>
<p>However, the naive double-dipping method will return many false
positives:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sel.dd</span> <span class="op">=</span> <span class="fu"><a href="../reference/dd.html">dd</a></span><span class="op">(</span><span class="va">simdata_1ct</span><span class="op">$</span><span class="va">simu_sce</span>, params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>normalized_method <span class="op">=</span> <span class="st">"sct"</span>, pca.whiten <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: The following arguments are not used: norm.method</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">sel.dd</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 67</span></span></code></pre></div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.1.3 (2022-03-10)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices datasets  utils     methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] future_1.68.0          SplitClusterTest_0.1.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] spatstat.univar_3.0-1       spam_2.11-0                </span></span>
<span><span class="co">#&gt;   [3] systemfonts_1.1.0           plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;   [5] igraph_2.1.1                lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;   [7] sp_2.1-4                    splines_4.1.3              </span></span>
<span><span class="co">#&gt;   [9] listenv_0.9.1               scattermore_1.2            </span></span>
<span><span class="co">#&gt;  [11] GenomeInfoDb_1.30.1         ggplot2_3.5.1              </span></span>
<span><span class="co">#&gt;  [13] digest_0.6.37               htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt;  [15] fansi_1.0.6                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;  [17] tensor_1.5                  cluster_2.1.2              </span></span>
<span><span class="co">#&gt;  [19] ROCR_1.0-11                 globals_0.18.0             </span></span>
<span><span class="co">#&gt;  [21] matrixStats_1.4.1           pkgdown_2.2.0              </span></span>
<span><span class="co">#&gt;  [23] spatstat.sparse_3.1-0       colorspace_2.1-1           </span></span>
<span><span class="co">#&gt;  [25] ggrepel_0.9.6               textshaping_0.4.0          </span></span>
<span><span class="co">#&gt;  [27] xfun_0.55                   dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [29] RCurl_1.98-1.16             jsonlite_1.8.9             </span></span>
<span><span class="co">#&gt;  [31] progressr_0.14.0            spatstat.data_3.1-2        </span></span>
<span><span class="co">#&gt;  [33] survival_3.2-13             zoo_1.8-12                 </span></span>
<span><span class="co">#&gt;  [35] glue_1.8.0                  polyclip_1.10-7            </span></span>
<span><span class="co">#&gt;  [37] gtable_0.3.5                zlibbioc_1.40.0            </span></span>
<span><span class="co">#&gt;  [39] XVector_0.34.0              leiden_0.4.3.1             </span></span>
<span><span class="co">#&gt;  [41] DelayedArray_0.20.0         future.apply_1.20.1        </span></span>
<span><span class="co">#&gt;  [43] SingleCellExperiment_1.16.0 BiocGenerics_0.40.0        </span></span>
<span><span class="co">#&gt;  [45] abind_1.4-8                 scales_1.3.0               </span></span>
<span><span class="co">#&gt;  [47] spatstat.random_3.3-2       miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt;  [49] Rcpp_1.0.13                 viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt;  [51] xtable_1.8-4                reticulate_1.39.0          </span></span>
<span><span class="co">#&gt;  [53] dotCall64_1.2               stats4_4.1.3               </span></span>
<span><span class="co">#&gt;  [55] htmlwidgets_1.6.4           httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [57] RColorBrewer_1.1-3          Seurat_4.4.0               </span></span>
<span><span class="co">#&gt;  [59] ica_1.0-3                   pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [61] farver_2.1.2                sass_0.4.9                 </span></span>
<span><span class="co">#&gt;  [63] uwot_0.2.2                  deldir_2.0-4               </span></span>
<span><span class="co">#&gt;  [65] utf8_1.2.4                  tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [67] rlang_1.1.4                 reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [69] later_1.3.2                 munsell_0.5.1              </span></span>
<span><span class="co">#&gt;  [71] tools_4.1.3                 cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [73] cli_3.6.3                   generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [75] ggridges_0.5.6              evaluate_1.0.1             </span></span>
<span><span class="co">#&gt;  [77] stringr_1.5.1               fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [79] yaml_2.3.10                 ragg_1.5.0                 </span></span>
<span><span class="co">#&gt;  [81] goftest_1.2-3               knitr_1.51                 </span></span>
<span><span class="co">#&gt;  [83] fs_1.6.4                    fitdistrplus_1.2-1         </span></span>
<span><span class="co">#&gt;  [85] purrr_1.0.2                 RANN_2.6.2                 </span></span>
<span><span class="co">#&gt;  [87] pbapply_1.7-4               nlme_3.1-155               </span></span>
<span><span class="co">#&gt;  [89] mime_0.12                   compiler_4.1.3             </span></span>
<span><span class="co">#&gt;  [91] plotly_4.10.4               png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [93] spatstat.utils_3.1-0        tibble_3.2.1               </span></span>
<span><span class="co">#&gt;  [95] bslib_0.8.0                 stringi_1.8.4              </span></span>
<span><span class="co">#&gt;  [97] desc_1.4.3                  lattice_0.20-45            </span></span>
<span><span class="co">#&gt;  [99] Matrix_1.6-5                vctrs_0.6.5                </span></span>
<span><span class="co">#&gt; [101] pillar_1.9.0                lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt; [103] BiocManager_1.30.25         spatstat.geom_3.3-3        </span></span>
<span><span class="co">#&gt; [105] lmtest_0.9-40               jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [107] RcppAnnoy_0.0.22            data.table_1.16.2          </span></span>
<span><span class="co">#&gt; [109] cowplot_1.1.3               bitops_1.0-9               </span></span>
<span><span class="co">#&gt; [111] irlba_2.3.5.1               httpuv_1.6.15              </span></span>
<span><span class="co">#&gt; [113] patchwork_1.3.0             GenomicRanges_1.46.1       </span></span>
<span><span class="co">#&gt; [115] R6_2.5.1                    promises_1.3.0             </span></span>
<span><span class="co">#&gt; [117] renv_1.0.10                 KernSmooth_2.23-20         </span></span>
<span><span class="co">#&gt; [119] gridExtra_2.3               IRanges_2.28.0             </span></span>
<span><span class="co">#&gt; [121] parallelly_1.46.1           codetools_0.2-18           </span></span>
<span><span class="co">#&gt; [123] MASS_7.3-55                 SummarizedExperiment_1.24.0</span></span>
<span><span class="co">#&gt; [125] SeuratObject_5.0.2          sctransform_0.4.1          </span></span>
<span><span class="co">#&gt; [127] S4Vectors_0.32.4            GenomeInfoDbData_1.2.7     </span></span>
<span><span class="co">#&gt; [129] parallel_4.1.3              grid_4.1.3                 </span></span>
<span><span class="co">#&gt; [131] tidyr_1.3.1                 rmarkdown_2.30             </span></span>
<span><span class="co">#&gt; [133] MatrixGenerics_1.6.0        Rtsne_0.17                 </span></span>
<span><span class="co">#&gt; [135] spatstat.explore_3.3-3      Biobase_2.54.0             </span></span>
<span><span class="co">#&gt; [137] shiny_1.9.1</span></span></code></pre></div>
</div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://hohoweiya.xyz" class="external-link">Lijun Wang</a>, <a href="https://yingxinlin.github.io/" class="external-link">Yingxin Lin</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
